object-utils.go

```diff
@@ -73,27 +73,32 @@ func IsValidBucketName(bucket string) bool {
 // - Caret ("^")
 // - Grave accent / back tick ("`")
 // - Vertical bar / pipe ("|")
+//
```

Comment:
no `object-name` is "stricter" means more restrictions than `prefix`  ok, look at it this way - Valid object names are a subset of valid prefix names.

what you are suggesting is keep things as is and just add strings.HasSuffix() to isValidObjectName like this:
```
func IsValidObjectName(object string) bool {
        if len(object) > 1024 || len(object) == 0 {
                return false
        }
        if !utf8.ValidString(object) {
                return false
        }
        if strings.HasSuffix(object, slashSeparator) {
                false
        }
        // Reject unsupported characters in object name.                                                                                                                          
        return !strings.ContainsAny(object, "`^*|\\\"")
}

// IsValidObjectPrefix verifies whether the prefix is a valid object name.                                                                                                        
// Its valid to have a empty prefix.                                                                                                                                              
func IsValidObjectPrefix(object string) bool {
        // Prefix can be empty or "/".                                                                                                                                            
        if object == "" || object == "/" {
                return true
        }
        // Verify if prefix is a valid object name.                                                                                                                               
        return IsValidObjectName(object)

}
```

but notice that this won't work as IsValidObjectPrefix() will fail if the prefix ends with `/` because of the check in isValidObjectName